js/initcontext.go

```diff
@@ -317,6 +313,40 @@ func (i *InitContext) Open(ctx context.Context, filename string, args ...string)
 	return i.runtime.ToValue(string(data)), nil
 }
 
+func readFile(fileSystem afero.Fs, filename string) (data []byte, err error) {
+	defer func() {
+		if errors.Is(err, fsext.ErrPathNeverRequestedBefore) {
+			// loading different files per VU is not supported, so all files should are going
+			// to be used inside the scenario should be opened during the init step (without any conditions)
+			err = fmt.Errorf(
+				"open() can't be used under with files that weren't opened during initialization (__VU==0), path: %q",
+				filename,
+			)
+		}
+	}()
```

Comment:
Note(not actionable): Technically, this error will either happen on `afero.IsDir` or not at all. But it seems kind of strange to check it only there and also when/if we ever stop checking that as we move to [io/fs](#1079) it will be a bit confusing why it wasn't previously checked.

I am noting this for future reference only, don't do anything about it just based on that.

Reply:
:thinking: not sure if I get it. The error happens `afero.IsDir` and `afero.ReadFile`. Even if we move to the `io/fs` still should happen on the equivalent of the `afero.ReadFile`, no?

Reply:
When we move – yes, but when we move to io/fs,  `afero.IsDir` will *hopefully* be just dropped ;). 

What I am trying to say is that if `afero.IsDir` did not return `fsext.ErrPathNeverRequestedBefore` than `afero.Readfile` will not return it either ... or we have bugs I guess or some strange race condition.

I wondered for a bit if we can't just add `afero.IsDir` check in the already reimplemented `Open` , but I am not certain we always use the CacheOnReadFs and IIRC we don't so :shrug: